projects/fw_102_hw/src/main.c

```diff
@@ -0,0 +1,36 @@
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+// not sure how to make it work with struct inside main()
+// would appreciate feedback!
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void counter_timer_callback(SoftTimerId timer_id, void *context) {

```

Comment:
Since this is a private function, please make it `static` and prefix its name with `prv_`. (The `static` makes it not visible from other files; otherwise there would be issues if another file tried to define a `counter_timer_callback`. The `prv_` is just a convention.)